protocol/lavaprotocol/finalization_consensus.go

```diff
@@ -53,44 +65,31 @@ func GetLatestFinalizedBlock(latestBlock, blockDistanceForFinalizedData int64) i
 func (fc *FinalizationConsensus) String() string {
 	fc.providerDataContainersMu.RLock()
 	defer fc.providerDataContainersMu.RUnlock()
-	mapExpectedBlockHeights := fc.GetExpectedBlockHeights(10 * time.Millisecond) // it's not super important so we hardcode this
-	return fmt.Sprintf("{FinalizationConsensus: {mapExpectedBlockHeights:%v} epoch: %d latestBlockByMedian %d}", mapExpectedBlockHeights, fc.currentEpoch, fc.latestBlockByMedian)
+	mapExpectedBlockHeights := fc.getExpectedBlockHeightsOfProviders(10 * time.Millisecond) // it's not super important so we hardcode this
+	return fmt.Sprintf("{FinalizationConsensus: {mapExpectedBlockHeights:%v} epoch: %d latestBlockByMedian %d}", mapExpectedBlockHeights, fc.currentEpoch, &fc.prevLatestBlockByMedian)
 }
 
-func (fc *FinalizationConsensus) newProviderHashesConsensus(blockDistanceForFinalizedData int64, providerAcc string, latestBlock int64, finalizedBlocks map[int64]string, reply *pairingtypes.RelayReply, req *pairingtypes.RelaySession) ProviderHashesConsensus {
+func (fc *FinalizationConsensus) insertProviderToConsensus(latestBlock, blockDistanceForFinalizedData int64, finalizedBlocks map[int64]string, reply *pairingtypes.RelayReply, req *pairingtypes.RelaySession, providerAcc string) {
 	newProviderDataContainer := providerDataContainer{
-		LatestFinalizedBlock:  GetLatestFinalizedBlock(latestBlock, blockDistanceForFinalizedData),
-		LatestBlockTime:       time.Now(),
-		FinalizedBlocksHashes: finalizedBlocks,
-		SigBlocks:             reply.SigBlocks,
-		SessionId:             req.SessionId,
-		RelayNum:              req.RelayNum,
-		BlockHeight:           req.Epoch,
-		LatestBlock:           latestBlock,
+		LatestFinalizedBlock:          GetLatestFinalizedBlock(latestBlock, blockDistanceForFinalizedData),
+		LatestBlockTime:               time.Now(),
```

Comment:
Spoke with @ranlavanet  about this, and agreed that this is actually ok. This is field is used to calculate the time difference between the `received` time of latest block, and the time which this container is used again. The calculated difference is used to determine the amount of blocks that has been built during this time window. There is no easy way to get the actual block time, and its is also not needed. The actual timestamps needed here is the receive time of this data, which is not too far from when the `time.Now()` is executed. This can't hurt the consumer whatsoever. Later on, we don't even use this number alone, but actually the median of all providers intact with the consumer at this epoch and the previous one.